docs repo documentation (#249)
metabase/docs.metabase.github.io: README.md

## Docs Workflow Overview

### [Happy Path](https://excalidraw.com/#json=FrrNqv5QUrjYIh5EyhzGJ,jLIOJkFupN0gPuHRMHdbOw)
![Docs Pipeline Happy Path](readme_assets/docs_pipeline_happy.excalidraw.png)

> ### How it works
> 
> We will examine the steps required to generate docs in the
> [docs.metabase.github.io](https://github.com/metabase/docs.metabase.github.io)
> repo.

Reviewer: ```suggestion
### How this repo builds docs from the Metabase repo

```

| Parameter | Required | Type | Default | Description |
|-----------------|----------|--------|-----------------|-------------------------------------------------------------------------------------|
| `source_branch` | ✅ | string | - | The feature/source branch from metabase/metabase that triggered this workflow |
| `target_branch` | ✅ | string | - | The target branch in metabase/metabase (`master` or `release-x.MM.x` for backports) |
| `annotation` | ❌ | string | `"auto-build"` | comment displayed in the build queue and PR title for data lineage |
| `dispatch_type` | ❌ | choice | `"docs_update"` | Action type: `docs_update`, `docs_merge`, or `docs_close` |
| `pr_number` | ❌ | string | `-` | adds a link to the docs PR to the original PR in metabase/metabase |
| `update_dirs` | ❌ | string | `""` | comma delimited string of extra dirs to git add during `docs_update` |

#### Tips and Best Practices

##### Annotations
- Use descriptive annotations to identify workflows in the build queue and PR
  titles
- Keep them concise since they appear in PR titles

##### Branch Naming
- Source branch must match exactly what's in metabase/metabase
- Target branch is typically `master`

> - Use `release-x.MM.x` format for backports (e.g., `release-0.50.x`)

Reviewer: ```suggestion
- Use `release-x.MM.x` format for backports (e.g., `release-x.50.x`)
```

- The source branch must exist in metabase/metabase when running this workflow.
  If the source branch has been deleted (e.g., after merging the original PR),

>   the workflow will fail.

Reviewer: >  If the source branch has been deleted (e.g., after merging the original PR),
  the workflow will fail.

What do we do in this scenario?

Author: ```suggestion
  If the source branch has been deleted (e.g., after merging the original PR),
  the workflow will fail. So follow the link to the main repo PR, and restore
  the branch there.
```

**Solution:**:
1. **Restore the branch** in metabase/metabase before running the workflow

### Branch Mapping
The workflow maps branches from the main repository to documentation changes:
- **Source → Target**: `docs-update-naming-55` → `master` creates a PR with
  branch `docs-update-naming-55->master` targeting `master` in this repo.

> - **Backports**: `docs-hotfix-security` → `release-0.50.x` creates a PR with

Reviewer: ```suggestion
- **Backports**: `docs-hotfix-security` → `release-x.50.x` creates a PR with
```

  branch `docs-hotfix-security->release-0.50.x` targeting `master` in this repo.

### Dispatch Types

#### `docs_update` (Default)
Creates or updates a documentation PR with the naming pattern `[annotation]
{source-branch}->{target-branch}`. The branchname for the PR will always be
`{source-branch}->{target-branch}`.

> **Example**: Source `docs-new-way-to-query` targeting `master` creates/updates
> PR named `[auto-build]docs-new-way-to-query->master`. This PR, in this repo,
> will have branchname `docs-new-way-to-query->master` and target `master` in this
> repo.

Reviewer: PRs or branches?

```suggestion
**Example**: 

In the Metabase repo, someone opens a PR. For example, a branch called `docs-edit`. The PR targets the `master` branch.

The Metabase PR will trigger this repo to create a new branch in this repo called `docs-edit->master` and a corresponding PR for that branch called `[auto-build]docs-edit->master` in this repo.
```

Does this cover it?

Author: ```suggestion
**Example**: 
	
	In the Metabase repo, someone opens a PR. For example, from a branch called `docs-edit` targeting `master`.
	
	The Metabase PR will trigger the `process_docs_changes` workflow in this repo to create a new PR in this repo called `docs-edit->master` and a corresponding PR for that branch named `[auto-build] docs-edit->master` in this repo.
```

| docs-workflow-test-1 | 0 |
| anything-else | 1 |

##### `cleanup_cloud_docs.clj`

> Cloud docs have been moved into the main repo! This script makes sure they only
> land in latest. Why? Because cloud always runs the latest version of Metabase.

Reviewer: ```suggestion
Makes sure the Cloud docs are only in the latest docs, as the cloud docs aren't versioned.
```
